### PR TITLE
Added LPM and reset support to sfputil for MLNX msn2700 platform [NOT FOR master]

### PR DESCRIPTION
**Pull request by andrii-savka**
**For review ONLY**
**PR can be merged after integrating MLNX SDK version 4.2.7000X042**
**Also the support for other MLNX platform will be added very soon**
**Currently support is available only for SN2700** 

**- What I did**
Added LPM and reset support to sfputil for MLNX platforms 

**-How I did it**
Implemented additional scripts which uses a MLNX SDK

**- How to verify it**
1. For reset - check the admin status during restart
2. For LPM:
2.1 Get LPM status
2.2 Set LPM to OFF
2.3 Get LPM status and check if it's OFF
2.4 Set LPM to ON
2.5 Get LPM status and check if it's ON

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmget.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+meta.swid = 0
+meta.access_cmd = SXD_ACCESS_CMD_GET
+
+rc = sxd_access_reg_mcion(mcion, meta, 1, None, None)
+assert rc == SXD_STATUS_SUCCESS, "sxd_access_reg_mcion failed, rc = %d" % rc
+
+# Get low power mode status
+lpm_mask = 1 << 8
+lpm_status = (lpm_mask & mcion.module_status_bits) != 0
+print "LMP ON" if lpm_status else "LPM OFF"
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmget.py]: `LMP ON` appears to be a typo. Should be `LPM ON`.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    print "Failed to open api handle.\nPlease check that SDK is running."
+    sys.exit(errno.EACCES)
+
+pid = os.getpid()
+rc = sxd_access_reg_init(pid, None, 0)
+if (rc != 0):
+    print "Failed to initializing register access.\nPlease check that SDK is running."
+    sys.exit(errno.EACCES);
+
+# Get SFP module and log ports number and LMP status
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: `LMP` should be `LPM`

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfpreset.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    print "Failed to open api handle.\nPlease check that SDK is running."
+    sys.exit(errno.EACCES)
+
+pid = os.getpid()
+rc = sxd_access_reg_init(pid, None, 0)
+if (rc != 0):
+    print "Failed to initializing register access.\nPlease check that SDK is running."
+    sys.exit(errno.EACCES)
+
+# Get SFP module number and LMP status
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfpreset.py]: `LMP` should be `LPM`

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 4]
```diff
@@ -65,13 +66,51 @@ def get_presence(self, port_num):
...
+        # Check for invalid port_num
+        if port_num < self.port_start or port_num > self.port_end:
+            return False
 
-        raise NotImplementedError
+        lpm_cmd = "docker exec syncd python /usr/share/sonic/platform/plugins/sfplpmget.py {}".format(port_num)
+
+        try:
+            output = subprocess.check_output(lpm_cmd, shell=True)
+            if 'LMP ON' in output:
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: `LMP ON` should be `LPM ON` (see my review comment on sfplpmget.py above).

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 5]
```diff
@@ -65,13 +66,51 @@ def get_presence(self, port_num):
...
+                return True
+        except subprocess.CalledProcessError as e:
+            print "Error! Unable to get LMP for {}, rc = {}, err msg: {}".format(port_num, e.returncode, e.output)
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: `LMP` should be `LPM`

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 6]
```diff
@@ -65,13 +66,51 @@ def get_presence(self, port_num):
...
+            return False
+
+        lpm = 'on' if lpmode else 'off'
+        lpm_cmd = "docker exec syncd python /usr/share/sonic/platform/plugins/sfplpmset.py {} {}".format(port_num, lpm)
+
+        try:
+            subprocess.check_output(lpm_cmd, shell=True)
+            return True
+        except subprocess.CalledProcessError as e:
+            print "Error! Unable to set LMP for {}, rc = {}, err msg: {}".format(port_num, e.returncode, e.output)
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: `LMP` should be `LPM`

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py [hunk 7]
```diff
@@ -65,13 +66,51 @@ def get_presence(self, port_num):
...
+            return False
 
-        raise NotImplementedError
+        lpm_cmd = "docker exec syncd python /usr/share/sonic/platform/plugins/sfpreset.py {}".format(port_num)
+
+        try:
+            subprocess.check_output(lpm_cmd, shell=True)
+            return True
+        except subprocess.CalledProcessError as e:
+            print "Error! Unable to set LMP for {}, rc = {}, err msg: {}".format(port_num, e.returncode, e.output)
```
Comment (jleveque) [inline device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: `LMP` should be `LPM`

**Review by jleveque — CHANGES_REQUESTED** [review]

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmget.py (hunk 1 above)
Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmget.py]: The typo is fixed.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py (hunk 2 above)
Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfplpmset.py]: The typo is fixed. The typo is fixed.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfpreset.py (hunk 3 above)
Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfpreset.py]: The typo is fixed.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py (hunk 4 above)
Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: The typo is fixed.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py (hunk 6 above)
Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: The typo is fixed.

File: device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py (hunk 7 above)
Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: The typo is fixed.

Comment (andrii-savka) [reply device/mellanox/x86_64-mlnx_msn2700-r0/plugins/sfputil.py]: The typo is fixed.

**Review by jleveque — APPROVED** [review]

**Comment by andrii-savka** [comment]
It can be merged after #13

**Comment by andrii-savka** [comment]
New issue with LPM setting is found. Don't merge this PR without trigger.

**Comment by andrii-savka** [comment]
PR #10 can be merged after #13

**Review by jleveque — APPROVED** [review]

**Comment by andrii-savka** [comment]
@lguohan @jleveque 
Rebased to the latest v1.2

**Pull request merged**